CMakeLists.txt

```diff
@@ -189,7 +189,8 @@ if (${CMAKE_CXX_COMPILER_ID} MATCHES Clang)
     # It seems this check should only be used for interopability with swift
     target_compile_options(hikogui INTERFACE -Wno-nullability-completeness)
 
-    target_compile_options(hikogui INTERFACE -march=${HI_ARCHITECTURE})
+    # TODO remove hardcoded architecture and use auto-detected ${HI_ARCHITECTURE}.
```

Comment:
The architecture need to be overridable as a developer should be able to select on what computers his application can run on. This computer will be different from the computer the build is done on.

Reply:
Ok. Build option is back in place. The HI_ARCHITECTURE value is now correctly set to the value of CPUINFO_ARCHITECTURE_LEVEL, if no value was provided on the CLI for this build option.